docs/architecture_overview.md

```diff
@@ -0,0 +1,130 @@
+
+Everything in the data model is backed inside of the `MephistoDB`, such that other processes can examine the status of tasks, view worker behavior, or access the information long after there is no active process running a crowdsourcing task. 
+## Runs
+Three primary classes talk about runs - namely the `Project`, `Task`, and `TaskRun`.  They are containers for work at varying levels of similarity.
+
+### Tasks
+To speak about what the `Task` class is for, first let's define a *Task* as a single type of work that you want to collect, and may do so in one or many chunks. In general, you'd want everything in a *Task* to be something that could be reviewed at the same time, and as such it should share similar context. 
+
+*Annotate [these] images* would be an example of a *Task*, as would *Translate  [these] sentences into [language]*. Each of these could be extended, for example by giving multiple (similar) annotation types or translating into many different languages. These two tasks though could not reasonably be combined into one *Task* though, as they are using different context. 
+
+Methods of the `Task` class surround being able to consolidate information about a specific *Task*, and the task class is uniquely indexed by the `task_name` property. 
+
+### Task Runs
+A task run can be thought of as part of running an overall task. So, if you wanted to annotate a set of images, but could only do it in parts, each of those parts would be a `TaskRun` of the image annotation `Task` that you are working on. Each `TaskRun` is provided its own set of configuration, and as such your `Task` to translate some number of sentences into different languages may have a `TaskRun` that asks workers to translate to Spanish while another `TaskRun` asks workers to translate the same sentences to French.
```

Comment:
It'd be wonderful if the previous example was extended with numbers (e.g. if you need to classify 100 images 3 times) how many `TaskRun` Mephisto will have, or just explain what does (parts) mean in this sentence (could classify images but in parts)